firebase/src/firebase.cpp

```diff
@@ -63,7 +64,7 @@ static int Firebase_Init(lua_State* L) {
 	JNIEnv* env = 0;
 	dmGraphics::GetNativeAndroidJavaVM()->AttachCurrentThread(&env, NULL);
 	
-	if (lua_isnil(L, 1))

```

Comment:
I think you could swap this for lua_isnoneornil(L, 1)

"Returns 1 if the given acceptable index is not valid (that is, it refers to an element outside the current stack) or if the value at this index is nil, and 0 otherwise."